x/epochs/types/hooks.go

```diff
@@ -13,29 +16,41 @@ type EpochHooks interface {
 	AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error
 	// new epoch is next block of epoch end block
 	BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error
+	// Returns the name of the module implementing epoch hook.
+	GetModuleName() string
 }
 
+const (
+	// flag indicating whether this is a before epoch hook
+	isBeforeEpoch = true
+)
+
 var _ EpochHooks = MultiEpochHooks{}
 
 // combine multiple gamm hooks, all hook functions are run in array sequence.
 type MultiEpochHooks []EpochHooks
 
+// GetModuleName implements EpochHooks.
+func (MultiEpochHooks) GetModuleName() string {
+	return ModuleName
+}
```

Comment:
Note: the core of the epoch modification is to expose this method for additional context with telemetry. It is state-compatible